### ISSUE TITLE
Document different backends on docker page

### ISSUE THREAD

**Issue by PatrickLemke** [mempool/mempool#386]
### Description

Currently there is no explanation on the docker page on how to use different backends.
It is defaulting to ELECTRUM, which is not automatically bundled with docker-compose, which means you already have to 
understand that you need ELECTRUM to get started.

#### Version

mempool/frontend            latest              3e5c9253d61f        3 weeks ago         101MB
mempool/backend             latest              680d65143073        3 weeks ago         294MB
mariadb                     10.5.8              ade39f0469a3        7 weeks ago         407MB

### Steps to reproduce

Follow instructions on docker markdown page

### Expected behaviour

ELECTRUM should either be installed automatically OR there should be documentation on how to select different backends.

### Actual behaviour

Logs show error messages because ELECTRUM is not connected.

**Comment by softsimon** [comment]
The project supports 3 type of backends:

1. blockstream/electrs (`"esplora"` as backend config)
2. bitcoind + romanz/electrs (`"electrum"` as backend config)
3. bitcoind (`"none"` as backend config)

I would like all to be available as Docker depending on user preference. Some users might just run Bitcoind and don't need the address lookup that electrum brings.

**Comment by wiz** [comment]
I agree we should make a docker-compose with the romanz/electrs image example so you can "one command install" mempool on any Linux system

**Comment by wiz** [comment]
Our newest contributor @RandyMcMillan finished the new Dockerfile and Makefile system, so now you can one-command install mempool, optionally also with electrs, but it would be nice to have someone who can write the documentation for this into our top-level README file or perhaps the docker/README file - @knorrium @emzy did you guys have any opinions on this?

**Comment by Emzy** [comment]
I was able to run the 2. type ("electrum" as backend config) with bitcoind and electrumx.
Works very well and fast.
electrumx: https://github.com/spesmilo/electrumx

**Comment by Emzy** [comment]
> ### Actual behaviour
> 
> Logs show error messages because ELECTRUM is not connected.

I had to build the backend (api) docker image myself. Because the image in the docker repository was missing the part to configure the electrum backend IP and PORT. So that fixed my problem, to be not able to connect to electrumx/electrs.

**Comment by knorrium** [comment]
> Our newest contributor @RandyMcMillan finished the new Dockerfile and Makefile system, so now you can one-command install mempool, optionally also with electrs, but it would be nice to have someone who can write the documentation for this into our top-level README file or perhaps the docker/README file - @knorrium @Emzy did you guys have any opinions on this?

@wiz I think there's more work to do on the Docker setup before we can document it.

Ideally we should be able to solve three main use cases:

1) Just running the latest published images (mostly done - I'm seeing some db permission issues and there's the issue mentioned above with the outdated backend image)
2) Building for local development (if someone wants to contribute to the Frontend, should they build everything or just the FE, pulling the latest backend image? What about the other way around? What if there are changes needed on both places? I think we should always build everything) 
3) Publishing images (done through Github Actions and we don't want to break it)

In order to fully support 1), we will need to improve the Makefile to take into account the user's preference for the backend. I started working on 2) so I could fix some bugs but I wasn't able to build the images locally.

**Comment by wiz** [comment]
1) wouldn't this be fixed as soon as we release v2.2, or is there additional fixes needed?
2) IMO probably best to just build both since the frontend takes up 95% of the build time
3) yeah I think our current GHA is fine for now, unless you want to add a master tag?

**Comment by knorrium** [comment]
> 1. wouldn't this be fixed as soon as we release v2.2, or is there additional fixes needed?
It's not working out of the box yet. I figured out what's causing the permissions issue I mentioned above (wrong path to the mysql volumes after we copy the docker-compose file to the root of the repo), but there are also some missing dependencies to build the electrum image.

I got it mostly working but I still can't connect:

```
pi_1       | Apr  1 06:02:37 [34] WARN: runMainLoop error: connect ECONNREFUSED 127.0.0.1:8332. Retrying in 10 sec.
api_1       | Apr  1 06:02:37 [34] DEBUG: {"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8332}
api_1       | Apr  1 06:02:42 [34] DEBUG: Electrum attempting reconnect...
api_1       | Apr  1 06:02:42 [34] ERR: Electrum error: {"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":50002}
api_1       | Apr  1 06:02:42 [34] ERR: Electrum error: {"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":50002}
api_1       | Apr  1 06:02:42 [34] INFO: Disconnected from Electrum Server at 127.0.0.1:50002
```

> 2. IMO probably best to just build both since the frontend takes up 95% of the build time
Agreed.

> 3. yeah I think our current GHA is fine for now, unless you want to add a master tag?
I think it's fine the way it is but now that we have a Makefile, we could have the tag and release targets there to have a single entry point to run, build and release all the different pieces.

**Comment by wiz** [comment]
@knorrium do you think you can finish in time for the v2.2.0 release?
